java/fury-core/src/main/java/org/apache/fury/memory/MemoryBuffer.java

```diff
@@ -1204,18 +1204,27 @@ public void writePrimitiveArray(Object arr, int offset, int numBytes) {
 
   /** For off-heap buffer, this will make a heap buffer internally. */
   public void grow(int neededSize) {
-    ensure(writerIndex + neededSize);
+    int length = writerIndex + neededSize;
+    if (length > size) {
+      growBuffer(length);
+    }
   }
 
   /** For off-heap buffer, this will make a heap buffer internally. */
   public void ensure(int length) {
     if (length > size) {
-      byte[] data = new byte[length * 2];
-      copyToUnsafe(0, data, Platform.BYTE_ARRAY_OFFSET, size());
-      initHeapBuffer(data, 0, data.length);
+      growBuffer(length);
     }
   }
 
+  private void growBuffer(int length) {
+    int newSize =
+        length < 104857600 ? length << 2 : (int) Math.min(length * 1.5d, Integer.MAX_VALUE);
```

Comment:
How is the value `104857600` calculated?

Reply:
It' just 100 * 1024 * 1024

Reply:
Move `org.apache.fury.io.FuryStreamReader#BUFFER_GROW_STEP_THRESHOLD` to `MemoryBuffer` and reuse it?

Reply:
good suggestion, updated